File: Application/Modules/AdminLTE/Common/Enum/SalesOrderType.cs
#region Enterprise Solution
// Copyright © 2006-2018 Enterprise Solution Limited
// Solution:  Enterprise Solution
// Project:   Enterprise Edition/Application
// File:  SalesOrderType.cs
// Date: 2018/04/03 20:51

#endregion
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using Microsoft.Common;
using Serenity.ComponentModel;

namespace Matrix
{
    [EnumKey("Enums.SalesOrderType")]
    public enum SalesOrderType
    {
        [Description("Standard")]
        Standard='S',
      
        [Description("Blanket")]
        Blanket='B',
        
        [Description("Release")]
        Release='R',
      
        [Description("Sample")]
        Sample='A'
    }
}

File: Application/Imports/ServerTypings/SalesOrderType.ts
namespace Matrix {
    export enum SalesOrderType {
        Sample = 65,
        Blanket = 66,
        Release = 82,
        Standard = 83
    }

    export class SalesOrderTypeOption  {
        Mapping: KeyedCollection<string>;
        constructor() {
        this.Mapping =new KeyedCollection<string>();
        this.Mapping.Add("A", "Sample");   //65,
        this.Mapping.Add("B", "Blanket");   //66,
        this.Mapping.Add("R", "Release");   //82,
        this.Mapping.Add("S", "Standard");   //83
        }
    }

    Serenity.Decorators.registerEnum(SalesOrderType, 'Enums.SalesOrderType');
}

